python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -494,21 +767,62 @@ def infer_schema_from_config_annotation(model_cls: Any, config_arg_default: Any)
 def infer_schema_from_config_class(
     model_cls: Type[Config],
     description: Optional[str] = None,
+    ignore_resource_fields: Optional[Set[str]] = None,
 ) -> Field:
     """
     Parses a structured config class and returns a corresponding Dagster config Field.
     """
+    ignore_resource_fields = ignore_resource_fields or set()
+
     check.param_invariant(
         issubclass(model_cls, Config),
         "Config type annotation must inherit from dagster._config.structured_config.Config",
     )
 
     fields = {}
     for pydantic_field in model_cls.__fields__.values():
-        fields[pydantic_field.alias] = _convert_pydantic_field(pydantic_field)
+        if pydantic_field.name not in ignore_resource_fields:
+            fields[pydantic_field.alias] = _convert_pydantic_field(pydantic_field)
 
     shape_cls = Permissive if model_cls.__config__.extra == Extra.allow else Shape
 
     docstring = model_cls.__doc__.strip() if model_cls.__doc__ else None
+    return Field(config=shape_cls(fields), description=description or docstring)
+
+
+def _separate_resource_params(
+    data: Dict[str, Any]
+) -> Tuple[Dict[str, Union[Resource, PartialResource, ResourceDefinition]], Dict[str, Any]]:
```

Comment:
nit: NamedTuple with property names could make this data structure more self-documenting